pyproject.toml

```
@@ -71,8 +71,8 @@ adlfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 gcsfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 psycopg2-binary = { version = ">=2.9.6", optional = true }
 sqlalchemy = { version = "^2.0.18", optional = true }
-numpy = { version = "^1.26.3", python = "3.12" }
-greenlet = {version = "^3.0.3", python = "3.12" }
+numpy = { version = "^1.26.3", python >= "3.12" }
+greenlet = {version = "^3.0.3", python >= "3.12" }
```

SORRY!  My siggestion here did NOT work so please revert.
```suggestion
numpy = { version = "^1.26.3", python = "3.12" }
greenlet = {version = "^3.0.3", python = "3.12" }
```

---

@MehulBatra Please make the above change so that the tests pass again.